### DISCUSSION TITLE
Weather segment configuration or font issue

### DISCUSSION THREAD

**Discussion by davidshen84** [erikw/tmux-powerline#442] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/cf89bf48-0fd0-474a-b02f-c5bb49166a62)

In my weather segment, there's a litter gap between the icon and the text. I am using the **CaskaydiaCove Nerd** font on my Windows Terminal app. I tried with other fonts, even the non-nerd fonts, and the gap is always there.

I found the temporary weather file:

```
☁️ 21.0°C
```

I found if I delete the weather emoji, there's an invisible block stay right at the same place. It's like the weather and the blank character overlay with each other. But when displaying on the status bar, the blank character causes a gap after the emoji.

I don't even know if it is an emoji bug or something is wrong with my Emacs.

**Comment by davidshen84** [comment]
I copied the emoji, ☁️, from https://emojipedia.org/cloud

In my Emacs, I found this:

- I put the cursor on the emoji and press delete. I got a blank character occupying that space
- I put the cursor after the emoji and press backspace. I got `☁`...which looks the same on the browser but looks smaller on my editor. In this case, the gap is gone and the status bar looks nice.

![image](https://github.com/user-attachments/assets/c4b972eb-a450-46aa-a286-a2ca86c1e2ef)

I think there's emoji trick here, but I just don't understand.

**Comment by davidshen84** [comment]
![image](https://github.com/user-attachments/assets/39c8bdde-45e4-410c-9587-c05c1995626f)

After I install the `emojify` emacs package, I can see there's indeed a blank square after some of the emoji characters.

**Comment by xx4h** [comment]
Hi,

thanks for reporting this!

yes, you're right, we investigated this already some weeks ago when making the code conform with the bash format checker.
This may also be related to: #351 

I'll have another look the coming days, thanks again for the report :+1:

**Comment by erikw** [comment]
Hey @davidshen84 did you resolve this issue? :)

**Reply by davidshen84** [reply, marked as answer]
I think it is resolved. I don't see the strange emoji gap now.
